### PR TITLE
- include 'empty' => true, 'default' => '' for date DEFAULT NULL fields

### PR DESCRIPTION
**Pull request by renardrouge**
Passed all current tests (11) under /bake

### CODE REVIEW

File: src/Template/Bake/Element/form.ctp [hunk 1]
```diff
@@ -71,9 +71,17 @@ $fields = collection($fields)
...
+                if (($fieldData['type'] === 'date') && (!empty($fieldData['null']))) {
+%>
+            echo $this->Form->input('<%= $field %>', array('empty' => true, 'default' => ''));
+<%
+                } else {
 %>
             echo $this->Form->input('<%= $field %>');
 <%
+                }
+                continue;
```
Comment (markstory) [inline src/Template/Bake/Element/form.ctp]: Why continue?

Comment (renardrouge) [inline src/Template/Bake/Element/form.ctp]: Modeled after block just previous, beginning    if (isset($keyFields[$field])) {

Comment (markstory) [inline src/Template/Bake/Element/form.ctp]: ok.

Comment (renardrouge) [inline src/Template/Bake/Element/form.ctp]: Hey Mark !
Something is not right.  That block of code in your "Why continue" question is not the same as the code in github.com/renardrouge/bake .

Comment (renardrouge) [inline src/Template/Bake/Element/form.ctp]: Oops !

I don't know what just happened.  I refreshed the screen, now the quoted code is the same as mine.

**Comment by dereuromark** [comment]
This seems to be missing for datetime, though.

**Pull request merged**